packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -408,7 +408,22 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
             )
           }
           needQueryInjectHelper = true
-          str().overwrite(start, end, `__vite__injectQuery(${url}, 'import')`)
+          if (
+            !/^('.*'|".*"|`.*`)$/.test(url) ||
+            (!isJSRequest(cleanUrl(url.slice(1, -1))) &&
```

Comment:
**suggestion (non-blocking):** We may extract `url.slice(1, -1)` to a local variable, so it is not called twice.
But this could make readability worse 🤔

Reply:
```suggestion
          const unwrappedUrl = /^('.*'|".*"|`.*`)$/.test(url) && url.slice(1, -1)
          if (
            !unWrappedUrl ||
            (!isJSRequest(cleanUrl(unwrappedUrl)) &&
              !isCSSRequest(unwrappedUrl))
          ) {
```
... maybe something like this?

Reply:
Yeah 🙂 I like that 👍
It even use falsy check if `unwrappedUrl` is an empty string

In addition, I'm thinking about if it is okay to use `.+` instead of `.*`... _But I need more ☕ for that 🤣_

Reply:
I mark this as resolved, because the following requested changes cover the same lines.